pandas/tests/io/test_excel.py

```diff
@@ -1058,351 +1008,291 @@ def test_read_excel_squeeze(self):
         tm.assert_series_equal(actual, expected)
 
 
-class TestXlsReaderTests(XlrdTests):
-    ext = '.xls'
-    engine_name = 'xlrd'
-    check_skip = staticmethod(_skip_if_no_xlrd)
-
-
-class TestXlsxReaderTests(XlrdTests):
-    ext = '.xlsx'
-    engine_name = 'xlrd'
-    check_skip = staticmethod(_skip_if_no_xlrd)
-
-
-class TestXlsmReaderTests(XlrdTests):
-    ext = '.xlsm'
-    engine_name = 'xlrd'
-    check_skip = staticmethod(_skip_if_no_xlrd)
-
-
-class ExcelWriterBase(SharedItems):
+class _WriterBase(SharedItems):
+    """Provides fixture to set / reset options for all writer tests"""
+
+    @pytest.fixture(autouse=True)
+    def set_options(self, request, merge_cells, engine, ext):
+        option_name = 'io.excel.{ext}.writer'.format(ext=ext.strip('.'))
+        prev_engine = get_option(option_name)
+        set_option(option_name, engine)
+        with ensure_clean(ext) as path:
```

Comment:
Because every `write` test uses `ensure_clean` as a context manager I figured it made more sense to add to the fixture rather than duplicating for each test

Reply:
totally fine. can you add some comments in the set_options, e.g. also a line explaining that this automatically used